rmc/utils/generic.py

```diff
@@ -497,77 +509,194 @@ def get_constraint_transcripts(outlier: bool = True) -> hl.expr.SetExpression:
         )
 
-        logger.info("Getting gene information from ClinVar HT...")
-        clinvar_ht = clinvar_ht.annotate(gene=clinvar_ht.info.GENEINFO.split(":")[0])
-        clinvar_ht = clinvar_ht.filter(hi_gene_set.contains(clinvar_ht.gene))
-        clinvar_ht = clinvar_ht.checkpoint(clinvar_ht_path, overwrite=overwrite)
+        logger.info("Filtering to variants in triplosensitive genes...")
+        triplo_genes = hl.experimental.read_expression(triplo_genes_he)
+        triplo_ht = ht.filter(triplo_genes.contains(ht.gene))
+        triplo_ht = triplo_ht.checkpoint(clinvar_plp_trip_mis.path, overwrite=overwrite)
         logger.info(
-            "Number of variants after filtering to HI genes: %i", clinvar_ht.count()
+            "Number of variants after filtering to triplosensitive genes: %i",
+            clinvar_ht.count(),
         )
 
 
-def import_de_novo_variants(
-    overwrite: bool,
-    csq_str: str = "consequence",
-    missense_str: str = MISSENSE,
-) -> None:
+def import_fu_data(overwrite: bool) -> None:
+    """
+    Import de novo variants from Fu et al. (2022) paper.
+
+    Function imports variants from TSV into HT, removes malformed rows,
+    and lifts data from GRCh38 to GRCh37.
+
+    :param overwrite: Whether to overwrite Table if it exists.
+    :return: None; Function writes Table to temporary path.
+    """
+    fu_ht = hl.import_table(
+        asc_ssc_spark_de_novo_tsv,
+        impute=True,
+        # Skip blank lines at the bottom of this TSV
+        missing="",
+        skip_blank_lines=True,
+    )
+    # Remove lines from bottom of TSV that are parsed incorrectly upon import
+    # These lines contain metadata about the TSV, e.g.:
+    # "Supplementary Table 20. The de novo SNV/indel variants used in TADA
+    # association analyses from assembled ASD cohorts"
+    fu_ht = fu_ht.filter(hl.is_missing(fu_ht.Role))
+
+    # Prepare to lift dataset back to GRCh37
+    rg37 = hl.get_reference("GRCh37")
+    rg38 = hl.get_reference("GRCh38")
+    rg38.add_liftover(
+        "gs://hail-common/references/grch38_to_grch37.over.chain.gz", rg37
+    )
+
+    fu_ht = fu_ht.annotate(
+        new_locus=hl.liftover(fu_ht.locus, "GRCh37", include_strand=True),
+        old_locus=fu_ht.locus,
+    )
+    liftover_stats = fu_ht.aggregate(
+        hl.struct(
+            liftover_fail=hl.agg.count_where(hl.is_missing(fu_ht.new_locus)),
+            flip_strand=hl.agg.count_where(fu_ht.new_locus.is_negative_strand),
+        )
+    )
+    logger.info(
+        "%i variants failed to liftover, and %i variants flipped strands",
+        liftover_stats.liftover_fail,
+        liftover_stats.flip_strand,
+    )
+    fu_ht = fu_ht.filter(
+        hl.is_defined(fu_ht.new_locus) & ~fu_ht.new_locus.is_negative_strand
+    )
+    fu_ht = fu_ht.key_by(locus=fu_ht.new_locus.result, alleles=fu_ht.alleles)
+    fu_ht = fu_ht.group_by("locus", "alleles").aggregate(
```

Comment:
We shouldn't be grouping these DNM tables by locus + allele as it would discard recurrence information, we just need to retain the phenotype labels and then concatenate the tables

Reply:
yes -- I did it this way because it was easier to store this smaller, de-duplicated version and simply explode on the sample set afterwards
<img width="718" alt="image" src="https://user-images.githubusercontent.com/19142091/219412232-0bb6c2b7-10a3-4704-901f-92f33c163c78.png">

Reply:
that's a good point, I'll switch to a `collect`

Reply:
I'm looking at this again and I think I'm ok with the collect aggregation, I think we can even slim this down more and remove `study` and `sample` and just leave the phenotype info ("DD", "Proband", "Sibling"). I think we should change "Proband" > "ASD" and "Sibling" > "Control" though for downstream ease of use.